lib/engine/game/g_18_cz.rb

```diff
@@ -167,6 +173,24 @@ def block_lay_for_purple_tiles
           tile.blocks_lay = true if tile.name.end_with?('p')
         end
       end
+
+      def must_buy_train?(entity)
+        !entity.rusted_self &&
+        !depot.depot_trains.empty? &&
+        (entity.trains.empty? ||
+          (entity.type == :medium && entity.trains.none? { |item| train_of_size?(item, :medium) }) ||
+          (entity.type == :large && entity.trains.none? { |item| train_of_size?(item, :large) }))
+      end
+
+      def train_of_size?(item, size)
+        name = if item.is_a?(Hash)
+                 item[:name]
+               else
+                 item.name
+               end
+
+        TRAINS_FOR_CORPORATIONS[size].include?(name)
```

Comment:
wouldn't it be better to invert this? 

TRAINS_FOR_CORPORATIONS[name] == size

Reply:
Maybe? I think the dictionary is easy to read with coropration size => trains but possible is to have a list of trains => corporation size.

I can change it but I don't really think one is much better than the other one.

Reply:
let's change it